seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/client/DorisSinkManager.java

```diff
@@ -0,0 +1,161 @@
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class DorisSinkManager {
+
+    private final SinkConfig sinkConfig;
+    private final List<byte[]> batchList;
+
+    private DorisStreamLoadVisitor dorisStreamLoadVisitor;
+    private ScheduledExecutorService scheduler;
+    private ScheduledFuture<?> scheduledFuture;
+    private volatile boolean initialize;
+    private volatile Exception flushException;
+    private int batchRowCount = 0;
+    private long batchBytesSize = 0;
+
+    private Integer batchIntervalMs;
+
+    public DorisSinkManager(SinkConfig sinkConfig, List<String> fileNames) {
+        this.sinkConfig = sinkConfig;
+        this.batchList = new ArrayList<>();
+        this.batchIntervalMs = sinkConfig.getBatchIntervalMs();
+        dorisStreamLoadVisitor = new DorisStreamLoadVisitor(sinkConfig, fileNames);
+    }
+
+    private void tryInit() throws IOException {
+        if (initialize) {
+            return;
+        }
+        initialize = true;
+
+        if (batchIntervalMs != null) {
+            scheduler = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("Doris-sink-output-%s").build());
+            scheduledFuture = scheduler.scheduleAtFixedRate(
+                () -> {
+                    try {
+                        flush();
+                    } catch (IOException e) {
+                        flushException = e;
+                    }
+                },
+                    batchIntervalMs,
+                    batchIntervalMs,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public synchronized void write(String record) throws IOException {
+        tryInit();
+        checkFlushException();
+        byte[] bts = record.getBytes(StandardCharsets.UTF_8);
+        batchList.add(bts);
+        batchRowCount++;
+        batchBytesSize += bts.length;
+        if (batchRowCount >= sinkConfig.getBatchMaxSize() || batchBytesSize >= sinkConfig.getBatchMaxBytes()) {
+            flush();
+        }
+    }
+
+    public synchronized void close() throws IOException {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+            scheduler.shutdown();
+        }
+
+        flush();
+    }
+
+    public synchronized void flush() throws IOException {
+        checkFlushException();
+        if (batchList.isEmpty()) {
+            return;
+        }
+        String label = createBatchLabel();
+        DorisFlushTuple tuple = new DorisFlushTuple(label, batchBytesSize, new ArrayList<>(batchList));
```

Comment:
new ArrayList<>(batchList)

what's the meaning of it? It can cause memory waste.

Reply:
The batchList is the data that the request needs to carry.

Reply:
This array list can be reused